### ISSUE TITLE
Revive old XML serialization tests

### ISSUE THREAD

**Issue by joleroi** [gammapy/gammapy#1402]
In the course of #1397 I xfailed all the old tests in ``gammapy/utils/tests/test_models.py``. They need to be reactivated and fixed, since the test usefull functionality.

**Comment by joleroi** [comment]
I will make a PR

**Comment by joleroi** [comment]
I noticed that this will be a larger project because it cannot be achieved with the current XML serialization module. The issue is that different XML parameter map to the same parameter name in gammapy, e.g.

* ``Prefactor`` :arrow_right: ``amplitude`` (PowerLaw)
* ``Integral`` :arrow_right:  ``amplitude`` (PowerLaw2)

Since the parameter names have a separate registry than the model names at the moment, there is no way to translate ``amplitude`` back to the correct XML name. I will think about a new scheme

**Comment by cdeil** [comment]
I see: https://github.com/gammapy/gammapy/blob/c45c3d9e807e37231e413edbaac2562118c977b7/gammapy/utils/serialization/xml.py#L37

How about deleting the parameter registry and putting something like this instead?
https://github.com/gammapy/gamma-cat/blob/master/model_serialisation/spectral_models.yaml

I think either a bundled YAML file, or just typing it in `xml.py` like you have now is fine: YAML and Python are very similar to read and write.

So that would introduce a bit of verbosity / duplicated info for parameters that appear for multiple models, but it would make the overall structure simpler by getting rid of the parameter registry.

@joleroi - What do you think?

**Comment by joleroi** [comment]
> How about deleting the parameter registry and putting something like this instead?

That would have been my first guess as well :+1:

**Comment by cdeil** [comment]
Just to note: the model XML I/O isn't in very good shape.
Especially the writing is completely broken at the moment.
(was it every working?)
But also most of the read tests are still xfail.

@joleroi - Do you have time for Gammapy next week?
If no, we could just defer this from v0.8 to v0.9.

**Comment by cdeil** [comment]
The registry data structure is designed for the case of reading XML, for the case of writing XML it's not well suited. This requires a bit more thought and work, so I'm postponing this to v0.9.

**Comment by adonath** [comment]
At the coding sprint we agreed, that for v1.0 (if at all) we will only support to read XML model files and not write and focus on a new YAML based prototype instead. For this issue it means that the tests for writing can be just removed and no time is spend to refactor it and make it work.

**Comment by cdeil** [comment]
Agreed that writing XML isn't very important for v1.0. On the other hand, what we want is to have a "translation layer" that allows the serialisation to be different from the Gammapy objects (e.g. class and parameter names), and this translation needs to be applied on read and write.

So if we develop this translation layer for the YAML serialisation, and design our serialisation API to support more than one format, then doing a partial support for XML would be the perfect test case to show that we have a good design. The main problem with the current code is that it uses this dict data structure that only allows translation in one direction.

So I would say: support for XML writing isn't required, but it's also not forbidden, especially if it helps to design and prototype a good solution. Basically: @QRemy - up to you how to proceed. Either remove some things now and then partially rebuild. Or try to refactor the existing code direction.

**Comment by registerrier** [comment]
No XML support within gammapy is envisioned now. Closing
